### ISSUE TITLE
Impossible to put a file with an inputStream, mandatory for large file

### ISSUE THREAD

**Issue by adiune-droid** [thegrizzlylabs/sardine-android#39]
This method is not implemented
void put(String url, InputStream dataStream, String contentType) throws IOException;

**Comment by guillaume-tgl** [comment]
That's right, see this ongoing PR or ideas on how to fix this. Feel free to open a new PR if you'd like to add support.

**Comment by Kitsune2222** [comment]
You can add OutputStream support? Seems http it support. How now upload files? I'm can't upload file by multiple **public void put(String url, byte[] data)** calls. 
Function **public void put(String url, File localFile, String contentType)** in some cases not possible for using, and no progress bar.
